File: php/features.php
<?php
declare(strict_types=1);

// StarWars SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class StarWarsFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new StarWarsBaseFeature();
            case "test":
                return new StarWarsTestFeature();
            default:
                return new StarWarsBaseFeature();
        }
    }
}

File: rb/features.rb
# StarWars SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module StarWarsFeatures
  def self.make_feature(name)
    case name
    when "base"
      StarWarsBaseFeature.new
    when "ratelimit"
      StarWarsRatelimitFeature.new
    when "retry"
      StarWarsRetryFeature.new
    when "test"
      StarWarsTestFeature.new
    when "timeout"
      StarWarsTimeoutFeature.new
    else
      StarWarsBaseFeature.new
    end
  end
end
